pyproject.toml

```diff
@@ -1,106 +1,174 @@
+authors = [{ name = "Hernan E. Grecco", email = "hernan.grecco@gmail.com" }]
+dynamic = ["version"]
+license = { text = "BSD" }
 description = "Physical quantities module"
 readme = "README.rst"
 maintainers = [
-  {name="Hernan E. Grecco", email="hernan.grecco@gmail.com"},
-  {name="Jules Chéron", email="julescheron@gmail.com"}
+  { name = "Hernan E. Grecco", email = "hernan.grecco@gmail.com" },
+  { name = "Jules Chéron", email = "julescheron@gmail.com" },
 ]
-keywords = ["physical", "quantities", "unit", "conversion", "science"]
 classifiers = [
-    "Development Status :: 4 - Beta",
-    "Intended Audience :: Developers",
-    "Intended Audience :: Science/Research",
-    "License :: OSI Approved :: BSD License",
-    "Operating System :: MacOS :: MacOS X",
-    "Operating System :: Microsoft :: Windows",
-    "Operating System :: POSIX",
-    "Programming Language :: Python",
-    "Topic :: Scientific/Engineering",
-    "Topic :: Software Development :: Libraries",
-    "Programming Language :: Python :: 3.9",
-    "Programming Language :: Python :: 3.10",
-    "Programming Language :: Python :: 3.11",
-    "Programming Language :: Python :: 3.12",
+  "Development Status :: 4 - Beta",
+  "Intended Audience :: Developers",
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: BSD License",
+  "Operating System :: MacOS :: MacOS X",
+  "Operating System :: Microsoft :: Windows",
+  "Operating System :: POSIX",
+  "Programming Language :: Python",
+  "Topic :: Scientific/Engineering",
+  "Topic :: Software Development :: Libraries",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+]
+requires-python = ">=3.11"
+keywords = ["physical", "quantities", "unit", "conversion", "science"]
+dependencies = [
+  "platformdirs>=2.1.0",
+  "typing_extensions>=4.0.0",
+  "flexcache>=0.3",
+  "flexparser>=0.4",
 ]
-requires-python = ">=3.9"
-dynamic = ["version", "dependencies"]
-
-[tool.setuptools.package-data]
-pint = [
-      "default_en.txt",
-      "constants_en.txt",
-      "py.typed"]
 
-[tool.setuptools.dynamic]
-dependencies = {file = "requirements.txt"}
+[tool.hatch.build]
+include = ["pint/default_en.txt", "pint/constants_en.txt"]
 
 [project.optional-dependencies]
-testbase = [
-    "pytest",
-    "pytest-cov",
-    "pytest-subtests",
-    "pytest-benchmark"
-]
-test = [
-    "pytest",
-    "pytest-mpl",
-    "pytest-cov",
-    "pytest-subtests",
-    "pytest-benchmark"
-]
-bench = [
-    "pytest",
-    "pytest-codspeed"
-]
+test = ["pytest", "pytest-cov", "pytest-subtests", "pytest-benchmark"]
+test-mpl = ["pytest-mpl"]
+test-full = ["pint[test, test-mpl]"]
+codspeed = ["pint[test-full]", "pytest-codspeed"]
 numpy = ["numpy >= 1.23"]
 uncertainties = ["uncertainties >= 3.1.6"]
 babel = ["babel <= 2.8"]
 pandas = ["pint-pandas >= 0.3"]
 xarray = ["xarray"]
 dask = ["dask"]
 mip = ["mip >= 1.13"]
+matplotlib = ["matplotlib"]
+full = [
```

Comment:
I'd use all, most packages use this terminology, 

See:

https://github.com/pandas-dev/pandas/blob/6bcd30397d67c3887288c7a82c2c235ce8bc3c7f/pyproject.toml#L85
https://github.com/pola-rs/polars/blob/947dc072724e8e6c7a0e9084fc97213e7643d526/py-polars/pyproject.toml#L82
https://github.com/fastapi/fastapi/blob/eea196f4a505c2584fafda691f2a0033c4db6622/pyproject.toml#L79

Reply:
Good point! I will change this